src/util_misc.cpp

```diff
@@ -386,7 +386,10 @@ int Util::readProcMapsLine(int mapsfd, ProcMapsArea *area)
   c = readDec (mapsfd, (VA *)&inodenum);
   area -> name[0] = '\0';
   while (c == ' ') c = readChar (mapsfd);
-  if (c == '/' || c == '[') { /* absolute pathname, or [stack], [vdso], etc. */
+  if (c == '/' || c == '[' || c == '(') {
+    // absolute pathname, or [stack], [vdso], etc.
+    // On some machines, deleted files have a " (deleted)" prefix to the
+    // filename.
```

Comment:
We don't have to check for the entire string. Even if something else comes up, the caller will handle the name appropriately. This particular piece of code just ensures that we don't simply fail on seeing a '(' at the beginning of a file name.

Having said that, if they were to introduce some other character, such as '{', we would be in trouble, but that seems quite unlikely and we should think about that only after we see something like that.